packages/filebrowser/src/browser.ts

```diff
@@ -68,6 +85,20 @@ export class FileBrowser extends SidePanel {
       this._trans.__('file browser')
     );
 
+    // Add a button to toggle the file filter.
+    this.toolbar.addItem(
+      'toggleFilter',
+      new ToolbarButton({
+        // Make icon vary based on state of filter?
+        icon: filterIcon,
+        onClick: () => {
+          this.toggleFileFilter();
+        },
+        tooltip: this._trans.__('Filter files by name'),
+        pressed: this._showFileFilter
+      })
+    );
```

Comment:
By hard-coding the button in the toolbar, this line prevents user- and administrator-configuration of the layout. Instead it would be great if a command could be used. Please note that if you use a command, the state returned by `isEnabled`, `isVisible` and `isToggled` will be automatically reflected and it will also help extension authors in case if they want to toggle it programatically or via a shortcut. Of note, all other buttons in this toolbar are already added via schema see:

https://github.com/jupyterlab/jupyterlab/blob/ae123cb82015540804964f76ba4e56f21799ea27/packages/filebrowser-extension/schema/widget.json#L4-L15

https://github.com/jupyterlab/jupyterlab/blob/ae123cb82015540804964f76ba4e56f21799ea27/packages/launcher-extension/schema/plugin.json#L24-L32